BlazarUI/app/scripts/components/module/BuildButton.jsx

```diff
@@ -0,0 +1,37 @@
+import React, {Component, PropTypes} from 'react';
+import {Button} from 'react-bootstrap';
+import Icon from '../shared/Icon.jsx';
+import {bindAll} from 'underscore';
+
+class BuildButton extends Component {
+
+  constructor(props) {
+    super(props);
+    bindAll(this, 'build');
+  }
+
+  build() {
+    this.props.triggerBuild();
+  }
+
+  render() {
+    if (this.props.buildTriggering) {
```

Comment:
when i try to build it flashes with the spinner and the disabled state, but then goes back to the normal state almost immediately. Also, we may need to get the current build state to determine if we want this disabled or not, rather than the buildTrigger. Say you land on this page and a build is in progress, the button should be disabled. or alternatively, some notification should show up saying a build is in progress and prevent the build.